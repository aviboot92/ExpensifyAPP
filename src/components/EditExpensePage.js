import React from 'react';

const EditExpensePage = (props) => {
    console.log(props);
    return(
        <div>
            <p>I am Edit Expense Page: {props.match.params.id}</p>
        </div>
    );
};

export default EditExpensePage;