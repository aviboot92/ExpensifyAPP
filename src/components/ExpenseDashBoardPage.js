import React from 'react';

const ExpenseDashBoardPage = () =>(
    <div>
        <p>I am Dash Board page</p>
    </div>
);

export default ExpenseDashBoardPage;