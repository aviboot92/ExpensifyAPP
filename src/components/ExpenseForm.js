import React from 'react'
import moment from 'moment';
import 'react-dates/initialize';
import {SingleDatePicker} from 'react-dates';


export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            descreption: props.expense? props.expense.descreption : "",
            amount: props.expense? (props.expense.amount/100).toString() :"",
            note: props.expense?props.expense.note:"",
            createdAt: props.expense? moment(props.expense.createdAt): moment(),
            calendarFocused: false,
            error: false
        }
    }

    onDescreptionChange = (e) => {
        const descreption = e.target.value;
        this.setState((prevState) => {
            return {descreption}
        });
    }

    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({amount}));
        }
    }

    onNoteChange = (e) => {
        e.persist();
        this.setState(() => ({note: e.target.value}));
    }

    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({createdAt}));
        }
    };

    onFocusChange = ({focused}) => {
        this.setState(() => ({calendarFocused: focused}));
    }

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.descreption || !this.state.amount) {
            this.setState((prevState) => ({
                ...prevState,
                error: true
            }))
        } else {
            if (this.state.error) {
                this.setState((prevState) => ({
                    ...prevState,
                    error: false
                }))
            }
            this
                .props
                .onSubmit({
                    descreption: this.state.descreption,
                    amount: parseFloat(this.state.amount, 10) * 100,
                    createdAt: this
                        .state
                        .createdAt
                        .valueOf(),
                    note: this.state.note
                });

        }
    }

    render() {
        return (
            <div>
                {this.state.error && <p>Please give right descreption and amount.</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Descreption"
                        autoFocus
                        value={this.state.descreption}
                        onChange={this.onDescreptionChange}/>
                    <input
                        type="text"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}/>

                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange
                        ={() => false}/>

                    <textarea
                        type="textArea"
                        placeholder="Note, It's just a descreption of your expense. (OPTIONAL!)"
                        value={this.state.note}
                        onChange={this.onNoteChange}></textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        );
    }
}