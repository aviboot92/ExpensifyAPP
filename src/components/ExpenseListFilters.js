import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilters extends React.Component {

    state = {
        calenderFocused: null
    }

    onDatesChange = ({ startDate, endDate }) => {
        this
            .props
            .dispatch(setStartDate(startDate));
        this
            .props
            .dispatch(setEndDate(endDate));
    }

    onFocusChange = (focusedInput) => {
        this.setState(() => {
            return { calenderFocused: focusedInput }
        })
    }
    
    render() {
        return (
            <div>
                <input
                    value={this.props.filters.text}
                    type="text"
                    onChange={(e) => {
                        this
                            .props
                            .dispatch(setTextFilter(e.target.value));
                    }} />
                <select
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                        (e.target.value === "amount")
                            ? this
                                .props
                                .dispatch(sortByAmount())
                            : this
                                .props
                                .dispatch(sortByDate());
                    }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    startDateId="your_unique_start_date_id"
                    endDate={this.props.filters.endDate}
                    endDateId="your_unique_end_date_id"
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calenderFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false} 
                    showClearDates = {true}/>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return { filters: state.filters }
}

export default connect(mapStateToProps)(ExpenseListFilters);