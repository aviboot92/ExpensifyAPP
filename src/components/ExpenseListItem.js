import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';

const ExpenseListItem = (props) => (
    <div>
        <p>Descreption :{props.descreption}</p>
        <p>Amount: {props.amount}</p>
        <p>Created At: {props.createdAt}</p>
        <button
            id={props.id}
            onClick={(e) => {
            props.dispatch(removeExpense({id: e.target.id}));
        }}>Remove</button>
        <br/>
    </div>
);

export default connect()(ExpenseListItem);