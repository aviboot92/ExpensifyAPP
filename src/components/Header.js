import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
    <header>
        <h1>Expensify APP</h1>
        <NavLink activeClassName='is-active' exact to='/'>Home</NavLink>
        <NavLink activeClassName='is-active' to='/create'>Add Expense</NavLink>
        <NavLink activeClassName='is-active' to='/help'>Help</NavLink>
    </header>
);

export default Header;