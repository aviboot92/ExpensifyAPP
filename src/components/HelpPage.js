import React from 'react';

const HelpPage = () => (
    <div>
        <p>I am Help Page</p>
    </div>
);

export default HelpPage;