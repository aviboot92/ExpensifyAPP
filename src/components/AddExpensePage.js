import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from "react-redux";
import {addExpense} from '../actions/expenses';

const AddExpensePage = (props) => (
    <div>
        <ExpenseForm 
          onSubmit={(expense) =>{
            props.addExpense({...expense});
              props.history.push("/");
          }}/>
    </div>
);

const mapDispatchToProps = {
    addExpense
}

export default connect(null, mapDispatchToProps)(AddExpensePage);