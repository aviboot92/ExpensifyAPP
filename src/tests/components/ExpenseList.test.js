import React from 'react';
import moment from 'moment';
import {shallow} from 'enzyme';
import {ExpenseList} from './../../components/ExpenseList';

const expenses = [
    {
        id: '1',
        descreption: 'Gum',
        note: '',
        amount: 100,
        createdAt: moment(0)
    }, {
        id: '2',
        descreption: 'Rent',
        note: '',
        amount: 10025,
        createdAt: moment(0)
            .subtract(4, 'days')
            .valueOf()
    }, {
        id: '3',
        descreption: 'Credit card',
        note: '',
        amount: 4500,
        createdAt: moment(0)
            .add(4, 'days')
            .valueOf()
    }
];

test('should render expenses correctly with expense',()=>{
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses correctly without expense',()=>{
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});