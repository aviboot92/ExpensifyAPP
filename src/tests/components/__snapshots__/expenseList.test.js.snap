// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={"1970-01-01T00:00:00.000Z"}
    descreption="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={19500000}
    createdAt={-345600000}
    descreption="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    descreption="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
